src/types.ts

```diff
@@ -1,5 +1,7 @@
 import { UploadType } from 'multinet';
 
+export type CSVColumnType = 'label' | 'category' | 'number' | 'date';
```

Comment:
Looking at the data types [here](https://www.arangodb.com/docs/stable/aql/fundamentals-data-types.html) to see what types we could natively store in Arango, it looks like an easy one to add would be boolean. However, if we're not really planning to leverage the native types, then this point is irrelevant.

Another quick note, some of these data types in the docs above would be better supported by a JSON upload (for example Arrays and Objects), if we do decide to go for some native storage in Arango.

Reply:
We have to be careful about this, because I think there's really two types flying around, which happen to intersect here. One is the actual data type that would be used to represent something in ArangoDB (JSON types: null, number, string, bool, etc.). The other is the "semantic" type, that is, how we choose to interpret the data in a column, regardless of it's actual data type. A good example of this might be if a a column is labeled as `date`, but each row entry could represent this with a number (unix timestamp) or a string (ISO 8601). Both would be valid ways to represent this data.

Maybe I'm incorrect about this, but regardless, I think more discussion is needed to flesh out the exact desired behavior. Thoughts @JackWilb @waxlamp?

Reply:
Right, thanks for explicitly spelling that out for us here. I agree that there are 2 different sets of types that we're talking about.

I guess the types that we use when storing data in Arango are actually irrelevant here since we can make a mapping from any type in the db to the types that we consider to be our "semantic" types.

Reply:
Good discussion. I like @AlmightyYakob's framing in terms of storage types and semantic types. And I like @JackWilb's observation that "semantic" types are more or less a hint to the vis app writer on what they may wish to do with the raw values coming from the database.

On the specific issue of a "bool" type: it is a good idea, but the difficulty is in execution. For instance, our heuristic could detect a column with only two values in it. Ordinarily, this would be marked as "categorical", but if the two values are strings `"true"` and `"false"` or `"yes"` and `"no"` then we might say it's a "bool" instead. Would we convert these values to literal bools in the database though? My gut says "no". For something like that, my intuition is that we leave it as "categorical" and let the client interpret the values. To have a first-class "bool" type we would really need to think about it (and we'd need a compelling use case first).

Good thoughts though!

Reply:
Oh, and on the topic of dates: one choice would be to do what BSON does, which is to serialize dates as this weird sort of structure:

```json
{
  "$date": {
    "value": "2020-10-16 16:41:21"
  }
}
```

(that's not precisely correct but it gets the idea across.)

If we sent up values like this from the client, that could serve as a decent signal that this is supposed to be a date.

I don't love this solution but our discussion made me think about BSON. So, just an idea to consider.

Reply:
> To have a first-class "bool" type we would really need to think about it (and we'd need a compelling use case first).

Just to follow up here, we do now have a compelling use case: UpSet. Let's talk about the boolean semantic type at our meeting on Wednesday.